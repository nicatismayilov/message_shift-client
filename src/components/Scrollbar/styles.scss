.scrollbar-thumb-vertical {
  background: #babac0;
  border-radius: 16px;
  cursor: pointer;
  -webkit-border-radius: 16px;
  -moz-border-radius: 16px;
  -ms-border-radius: 16px;
  -o-border-radius: 16px;
}

.scrollbar-thumb-horizontal {
  background: #babac0;
  border-radius: 16px;
  cursor: pointer;
  -webkit-border-radius: 16px;
  -moz-border-radius: 16px;
  -ms-border-radius: 16px;
  -o-border-radius: 16px;
}

.scrollbar-view {
  &-horizontal {
    display: flex;
    flex-direction: row;
  }

  &-vertical {
    display: flex;
    flex-direction: column;
  }
}
