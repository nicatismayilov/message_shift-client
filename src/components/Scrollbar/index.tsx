import React from "react";
import { Scrollbars } from "react-custom-scrollbars";

import "./styles.scss";

interface Props {
	hide?: boolean;
	renderHorizontal?: boolean;
}

const Scrollbar: React.FC<Props> = ({ hide = false, renderHorizontal = false, ...props }) => {
	return (
		<Scrollbars
			autoHide={hide}
			style={{
				height: "100%",
				width: "100%",
			}}
			renderThumbVertical={(props) => <div {...props} className='scrollbar-thumb-vertical' />}
			renderThumbHorizontal={(props) => <div {...props} className='scrollbar-thumb-horizontal' />}
			renderView={(props) =>
				renderHorizontal ? (
					<div {...props} className={`scrollbar-view-horizontal`} />
				) : (
					<div {...props} className={`scrollbar-view-vertical`} />
				)
			}
			hideTracksWhenNotNeeded={true}
		>
			{props.children}
		</Scrollbars>
	);
};

export default Scrollbar;
