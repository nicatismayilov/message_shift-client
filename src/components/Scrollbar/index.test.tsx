import React from "react";
import { render } from "@testing-library/react";

import Scrollbar from "./index";

describe("Scrollbar component", () => {
	it("should match the snapshot", () => {
		const { baseElement } = render(<Scrollbar />);

		expect(baseElement).toMatchSnapshot();
	});
});
