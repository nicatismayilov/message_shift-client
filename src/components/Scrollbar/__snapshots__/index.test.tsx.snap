// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scrollbar component should match the snapshot 1`] = `
<body>
  <div>
    <div
      style="position: relative; overflow: hidden; width: 100%; height: 100%;"
    >
      <div
        class="scrollbar-view-vertical"
        style="position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; overflow: scroll; margin-right: 0px; margin-bottom: 0px;"
      />
      <div
        style="position: absolute; height: 6px; display: none; right: 2px; bottom: 2px; left: 2px; border-radius: 3px;"
      >
        <div
          class="scrollbar-thumb-horizontal"
          style="position: relative; display: block; height: 100%;"
        />
      </div>
      <div
        style="position: absolute; width: 6px; display: none; right: 2px; bottom: 2px; top: 2px; border-radius: 3px;"
      >
        <div
          class="scrollbar-thumb-vertical"
          style="position: relative; display: block; width: 100%;"
        />
      </div>
    </div>
  </div>
</body>
`;
